### DISCUSSION TITLE
HTTPS e certificati attendibili

### DISCUSSION THREAD

**Discussion by gforment** [indomus/forum#262] [HUB: Home Assistant]
Ciao,
per diversi motivi ho dovuto fare una installazione da zero e sto approfittando per fare le cose in maniera più critica e documentarle (ps: se qualcuno è interessato sto tracciando passo passo una installazione su RPI4 con Docker per gestire i vari container, boot da SD Card ma tutto il resto cerco di tenerlo su un SDD).
Sto ripassando dalla guida InDomus per l'esposizione HTTPS su internet e creazione dei certificati. Dubbio: collegandomi a HA la connessione viene indicata come non attendibile (immagino legata al tema Certification Authority, ecc.) e ogni tanto  compare il classico Warning che richede conferma esplicita di voler proseguire. Esiste un modo per trustare i certificati o usare certificati emessi da una certification authority? Credo di aver provato in passato anche a mettere i certificati nei certificati trusted di Windows.
Al di là del fatto che ogni tanto sia necessario confermare l'attendibilità della connessione, quando provo ad inserire degli iframe di Grafana (sempre su HTTPS) anche questi chiaramente vanno in errore a meno di non aprire Grafana in un altro tab e confermare l'attendibilità anche per quella connessione.
Suggerimenti?
Grazie!

**Comment by snakuzzo** [comment]
Ciao, innanzi tutto devi indicare quale guida hai seguito e come hai esposto HA...tramite reverse proxy ? Direttamente ? Come ?
Poi...se ti dà quei warning hai sbagliato qualcosa. NON devi trustare alcun certificato. Se il certificato è OK...è ok. Da qui non si scappa. Ti consiglio di ricontrollare tutta la configurazione relativa all'HTTPS.
Per quanto riguarda le schermate in panel iframe...se esponi HA in https direttamente non potrai mai visualizzare nulla in http o in https con handshake che fallisce. Quindi per farlo dovrai utilizzare un reverse proxy a monte.

**Reply by gforment** [reply]
Guida InDomus https://indomus.it/guide/collegarsi-da-remoto-a-home-assistant-installato-su-raspberry-raspbian/
I certificati sono stati generati correttamente (come da "screenshot" della guida) e la configurazione è la seguente:
```
http:
  ssl_certificate: !secret ssl_certificate
  ssl_key: !secret ssl_key
```
La comunicazione HTTPS è attiva ma compare il warning.... ma giustamente non succede se mi collego anche da rete interna tramite URL duckdns invece che alias locale... quindi ho risolto! In questo caso è stato rubber duck debugging nel vero senso della parola... grazie!

In merito ad Iframe: HA e Grafana sono in HTTPS quindi HA può visualizzare e la visualizza correttamente ma solo dopo che il warning è stato confermato per quell'host. Quindi credo di aver risolto anche per quello ma forse va ingegnerizzato meglio per evitare esposizioni verso internet anche di Grafana

**Reply by snakuzzo** [reply]
Per non avere il warning devi collegarti usando il dominio. E' OVVIO che fallisca se ti colleghi con ip o altro nome...perché fallisce l'handshake.
Resta il concetto sugli iframe che ti ho scritto prima..da lì non si scappa. Lo vedi con warning...e questo NON va bene. Il certificato deve essere valido.
Questa cosa su HA Core si fa correttamente con un reverse proxy...ed è la stessa cosa che fa Hassio con i suoi addon.
Ad ogni modo ti consiglio di guardare la guida di Caddy su Docker
